code/modules/multiz/hoist.dm

```diff
@@ -0,0 +1,182 @@
+		user_unbuckle_mob(user)
+	source_hoist.hoistee.anchored = 0
+	source_hoist.hoistee = null
+	user.visible_message(span("warning", "[user] detaches \the [source_hoist.hoistee] from the hoist clamp."), span("notice", "You detach \the [source_hoist.hoistee] from the hoist clamp."), span("notice", "You hear something unclamp."))
+*/
+
+/obj/structure/hoist
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "pipe_d"
+	density = 1
+	anchored = 1
+	name = "Hoist"
+	desc = "A manual hoist, uses a clamp and pulley to hoist things."
+	var/atom/movable/hoistee
+	var/movedir = UP
+	var/obj/effect/hoist_hook/source_hook
+
+/obj/structure/hoist/Initialize(mapload, ndir)
+	. = ..()
+	dir = ndir
+	var/turf/newloc = get_step(src, dir)
+	source_hook = new(newloc)
+	source_hook.source_hoist = src
+
+/obj/structure/hoist/attack_hand(mob/living/user)
+	var/can = movedir == UP ? can_raise() : can_lower()
+	var/movtext = movedir == UP ? "raise" : "lower"
+	if (!can) // If you can't...
+		movedir = movedir == UP ? DOWN : UP // switch directions!
+		to_chat(user, span("notice", "You switch the direction of the pulley."))
+		return
+
+	if (!hoistee)
+		user.visible_message(span("notice", "[user] begins to [movtext] the clamp."), span("notice", "You begin to [movtext] the clamp."), span("notice", "You hear the sound of a crank."))
+		movedir == UP ? raise(0) : lower(0)
+		return
+
+	var/size
+	if (ismob(hoistee))
+		var/mob/M = hoistee
+		size = M.mob_size
+	else if (isobj(hoistee))
+		var/obj/O = hoistee
+		size = O.w_class
+
+	user.visible_message(span("notice", "[user] begins to [movtext] \the [hoistee]!"), span("notice", "You begin to [movtext] \the [hoistee]!"), span("notice", "You hear the sound of a crank."))
+	if (do_after(user, (1 SECONDS) * size / 4, act_target = src))
+		movedir == UP ? raise(1) : lower(1)
+
+/obj/structure/hoist/verb/collapse_hoist()
+	set name = "Collapse Hoist"
+	set category = "Object"
+	set src = view(1)
+
+	if (isobserver(usr))
+		return
+	if (!usr.IsAdvancedToolUser()) // thanks nanacode
+		to_chat(usr, span("notice", "You stare cluelessly at \the [src]."))
+		return
+
+	if (hoistee)
+		to_chat(usr, span("notice", "You cannot collapse the hoist with \the [hoistee] attached!"))
+		return
+	QDEL_NULL(src.source_hook)
+	new /obj/item/hoist_kit(get_turf(src))
+	qdel(src)
+
+/obj/structure/hoist/proc/can_raise()
+	var/turf/above = GetAbove(source_hook)
+	if (!above) // can't move if there's nothing to move to
+		return 0
+	if (!istype(above, /turf/simulated/open)) // can't move into a solid tile
+		return 0
+	if (source_hook in get_step(src, dir)) // you don't get to move above the hoist
+		return 0
+	return 1
+
+/obj/structure/hoist/proc/can_lower()
+	var/turf/below = GetBelow(source_hook)
+	if (!below) // can't move if there's nothing to move to
+		return 0
+	if (!istype(get_turf(source_hook), /turf/simulated/open)) // can't move down through a solid tile
+		return 0
+	return 1
+
+/obj/structure/hoist/proc/raise(ishoisting)
```

Comment:
Can you not collapse this and lower into a singular proc that takes `direction` as a parameter? It's awful code duplication otherwise.